opal/mca/btl/base/btl_base_am_rdma.c

```diff
@@ -0,0 +1,1064 @@
+
+static OBJ_CLASS_INSTANCE(mca_btl_base_rdma_operation_t, opal_list_item_t, NULL, NULL);
+
+static inline size_t size_t_min (size_t a, size_t b) {
+    return (a < b) ? a : b;
+}
+
+static inline size_t size_t_max (size_t a, size_t b) {
+    return (a > b) ? a : b;
+}
+
+static mca_btl_base_am_rdma_module_t default_module;
+
+static inline bool mca_btl_base_rdma_use_rdma_get (mca_btl_base_module_t *btl)
+{
+    return !!(btl->btl_flags & MCA_BTL_FLAGS_GET);
+}
+
+static inline bool mca_btl_base_rdma_use_rdma_put (mca_btl_base_module_t *btl)
+{
+    return !!(btl->btl_flags & MCA_BTL_FLAGS_PUT);
+}
+
+static inline bool mca_btl_base_rdma_is_atomic (mca_btl_base_rdma_type_t type) {
+    return (MCA_BTL_BASE_AM_PUT != type && MCA_BTL_BASE_AM_GET != type);
+}
+
+static inline
+size_t mca_btl_base_rdma_operation_size (mca_btl_base_module_t *btl, mca_btl_base_rdma_type_t type,
+                                         size_t remaining)
+{
+    switch (type) {
+    case MCA_BTL_BASE_AM_PUT:
+        if (mca_btl_base_rdma_use_rdma_get (btl)) {
+            return size_t_min (remaining, btl->btl_get_limit);
+        }
+        break;
+    case MCA_BTL_BASE_AM_GET:
+        if (mca_btl_base_rdma_use_rdma_put (btl)) {
+            return size_t_min (remaining, btl->btl_put_limit);
+        }
+        break;
+    case MCA_BTL_BASE_AM_ATOMIC:
+        /* fall through */
+    case MCA_BTL_BASE_AM_CAS:
+        return remaining;
+    }
+
+    return size_t_min (remaining, btl->btl_max_send_size - sizeof (mca_btl_base_rdma_hdr_t));
+}
+
+static inline int mca_btl_base_rdma_tag (mca_btl_base_rdma_type_t type) {
+    (void) type;
+    switch (type) {
+    case MCA_BTL_BASE_AM_PUT:
+    case MCA_BTL_BASE_AM_GET:
+        return MCA_BTL_BASE_TAG_RDMA;
+    case MCA_BTL_BASE_AM_ATOMIC:
+    case MCA_BTL_BASE_AM_CAS:
+        return MCA_BTL_BASE_TAG_ATOMIC;
+    }
+    return MCA_BTL_BASE_TAG_RDMA_RESP;
+}
+
+static inline int mca_btl_base_rdma_resp_tag (void) {
+    return MCA_BTL_BASE_TAG_RDMA_RESP;
+}
+
+/**
+ * @brief copy data from a segment to a local address
+ *
+ * @in addr          location to store data
+ * @in skip_bytes    bytes of segment data to skip before copying
+ * @in segments      segments to copy data from
+ * @in segment_count number of segments
+ */
+static void mca_btl_base_copy_from_segments (uint64_t addr, size_t skip_bytes,
+                                             const mca_btl_base_segment_t *segments,
+                                             size_t segment_count) {
+    const void *seg0_data = (const void *)((uintptr_t) segments[0].seg_addr.pval + skip_bytes);
+    size_t seg0_len = segments[0].seg_len - skip_bytes;
+
+    if (seg0_len > 0) {
+        BTL_VERBOSE(("unpacking %" PRIsize_t " bytes from segment 0 to %p", seg0_len, (void *) addr));
+        memcpy ((void *) addr, seg0_data, seg0_len);
+        addr += seg0_len;
+    }
+
+    for (size_t i = 1 ; i < segment_count ; ++i) {
+	size_t seg_len = segments[i].seg_len;
```

Comment:
Indentation in this block is off

Reply:
looks like tabs from before indent-tabs-mode:nil took effect. Fixed.